modules/iam-eks-role/data.tf

```diff
@@ -0,0 +1,8 @@
+data "aws_caller_identity" "current" {}
```

Comment:
Please move all these data sources at the top of `main.tf` like we have in all other modules.